src/auth/utils.ts

```diff
@@ -558,6 +563,81 @@ export class AuthNode implements TreeNode<Auth> {
     }
 }
 
+export async function credentialExists(
+    allConnections = () => Auth.instance.listAndTraverseConnections().promise()
+): Promise<boolean> {
+    return (await allConnections()).some(isIamConnection)
+}
+
+export type SsoKind = 'any' | 'codewhisperer'
+
+/**
+ * Returns true if an Identity Center SSO connection exists.
+ *
+ * @param kind A specific kind of Identity Center SSO connection that must exist.
+ * @param allConnections func to get all connections that exist
+ */
+export async function ssoExists(
+    kind: SsoKind = 'any',
+    allConnections = () => Auth.instance.listConnections()
+): Promise<boolean> {
+    return (await findSsoConnections(kind, allConnections)).length > 0
+}
+
+async function findSsoConnections(
+    kind: SsoKind = 'any',
+    allConnections = () => Auth.instance.listConnections()
+): Promise<SsoConnection[]> {
+    let predicate: (c?: Connection) => boolean
+    switch (kind) {
+        case 'codewhisperer':
+            predicate = (conn?: Connection) => {
+                return isBaseSsoConnection(conn) && isValidCodeWhispererConnection(conn)
+            }
+            break
+        case 'any':
+            predicate = isBaseSsoConnection
+    }
+    return (await allConnections()).filter(predicate).filter(isBaseSsoConnection)
+}
+
+export type BuilderIdKind = 'any' | 'codewhisperer' | 'codecatalyst'
+
+/**
+ * Returns true if a Builder ID connection exists.
+ *
+ * @param kind A Builder ID connection that has the scopes of this kind.
+ * @param allConnections func to get all connections that exist
+ */
+export async function builderIdExists(
```

Comment:
would `has` be a more discoverable prefix for these functions:

```
hasBuilderId
hasSso
hasIamCredentials
```

That also parallels nicely with the `isIamConnection`/`isSsoConnection`/etc functions.

Reply:
Updated